PlaceholderExpansion.md

```diff
@@ -71,265 +74,298 @@ Let's quickly break down the different methods you have to implement.
 
-You must choose between one of these two methods for handling the actual parsing of placeholders:
+You must choose between one of these two methods for handling the actual parsing of placeholders (Unless you're using [Relational Placeholders](#relational-placeholders)):
 
 - #### onRequest(OfflinePlayer, String)
   If not explicitly set, this will automatically call [`onPlaceholderRequest(Player, String)`](#onplaceholderrequestplayer-string).
-  This method is recommended as it allows the usage of offline players, meaning the player doesn't need to be online to get certain information (i.e. name).
+  This method is recommended as it allows the usage of offline players, meaning the player doesn't need to be online to obtain certain data from them like name or UUID.
 - #### onPlaceholderRequest(Player, String)
   If not set, this method will return `null` which PlaceholderAPI sees as an invalid placeholder.  
   The `Player` can be `null`, so keep that in mind when handling your placeholders.
 
+PlaceholderAPI will **always** call `onRequest(OfflinePlayer, String)` in an expansion.
+
 ----
-## Without a Plugin
-An expansion does not always need a plugin to rely on. If the placeholders it provides can return values from just the server itself or some other source (i.e. Java itself), then it can work independently.
 
-Common examples of such Expansions are:
+## Making an internal Expansion
 
-- [Player Expansion][playerexpansion]
-- [Server Expansion][serverexpansion]
-- [Math Expansion][mathexpansion]
+Internal PlaceholderExpansions are classes directly implemented into the plugin they depend on. The main benefit is, that you're not required to do any `canRegister()` check for your own plugin, as it is loaded within it.  
+Another benefit is that you can more easily access plugin data using Dependency Injection should an API not be accessible for the plugin.
 
-These kinds of expansions don't require any additional plugins to function.  
-When creating such an expansion is it recommended to use [`onRequest(OfflinePlayer, String)`](#onrequestofflineplayer-string).
+It is important to note that PlaceholderExpansions loaded manually through the `register()` method (Therefore not being loaded by PlaceholderAPI itself) **require** to have the `persist()` method return `true` to avoid the expansion being unloaded during a `/papi reload`.
 
-#### Full Example
-Please see the [Common parts](#common-parts) section for info on the other methods.
+Depending on what info you try to access from your plugin (i.e. through the main class or a dedicated API) can you use Dependency Injection to obtain an instance of whatever you need.  
+Keep in mind that as mentioned earlier, the Expansion class won't be reloaded when `persist()` is set to `true`, so you have to refresh any info that may get outdated during a reload yourself.
 
+Here is an example of a possible Dependency Injection:  
 ```java
```

Comment:
Some of these examples can get quite long -- perhaps surround with GitHub's collapsible detail elements to make reading this a little easier?

```
<details>
<summary>Example registration through dependency inejction</summary>

\```java
#foo
\```

</details>
```

Reply:
It does. Just need a space between the HTML tag and markdown

Reply:
Moved code examples and the explanation of the methods into details. I kinda dislike that details are barely noticable, but I think it should be fine.